test/CommunityVault.t.sol

```diff
@@ -38,10 +46,178 @@ contract CommunityVaultTest is Test {
         );
 
         vault = new CommunityVault(address(ownerToken), address(masterToken));
+
+        accounts[0] = makeAddr("one");
+        accounts[1] = makeAddr("two");
     }
 
     function test_Deployment() public {
         assertEq(vault.ownerToken(), address(ownerToken));
         assertEq(vault.masterToken(), address(masterToken));
     }
 }
+
+contract CommunityVaultBaseERC20Test is CommunityVaultTest {
+    function setUp() public virtual override {
+        CommunityVaultTest.setUp();
+
+        // mint 10 tokens to user
+        address user = accounts[0];
+        erc20Token.mint(user, 10e18);
+
+        // user transfer 10 tokens to the vault
+        vm.prank(user);
+        erc20Token.transfer(address(vault), 10e18);
+    }
+}
+
+contract TransferERC20ByNonAdminTest is CommunityVaultBaseERC20Test {
+    function setUp() public virtual override {
+        CommunityVaultBaseERC20Test.setUp();
+    }
+
+    function test_revertifcalledbynonadmin() public {
+        assertEq(erc20Token.balanceOf(address(vault)), 10e18);
+
+        uint256[] memory amounts = new uint256[](2);
+        amounts[0] = 1;
+        amounts[1] = 1;
+
+        vm.prank(accounts[0]);
```

Comment:
The test contract itself is going to be `msg.sender` by default, so no `vm.prank()` needed here.

Reply:
I would prefer to have it explicitly written in case we change the way we deploy in the future, wdyt?

Reply:
That's perfectly fine. Was not an actionable comment.